File: Lexical_Smallest_Palindrome/palindrome.py
class Solution:
    def makeSmallestPalindrome(self, s: str) -> str:
        char_list = list(s)

        i, j = 0, len(char_list) - 1

        while i < j:
            if char_list[i] != char_list[j]:
                char_list[i] = min(char_list[i], char_list[j])
                char_list[j] = char_list[i]

            i += 1
            j -= 1

        return "".join(char_list)

File: Lexical_Smallest_Palindrome/palindrome.go
package lexicalsmallestpalindrome

func min(a, b byte) byte {
	if a < b {
		return a
	}

	return b
}

func makeSmallestPalindrome(s string) string {
	charArr := []byte(s)

	for i, j := 0, len(charArr)-1; i < j; i, j = i+1, j-1 {
		if charArr[i] != charArr[j] {
			charArr[i], charArr[j] = min(charArr[i], charArr[j]), min(charArr[i], charArr[j])
		}
	}

	return string(charArr)
}
